File: kata/range/jython/range.py
from jarray import zeros, array

class Range: 
	def __init__(self, range_):
		se = range_[1:-1].split(",")
		self.start = int(se[0])
		if (range_[0] == "("):
			self.start = self.start + 1
		self.end = int(se[1])
		if (range_[-1] == ")"):
			self.end = self.end - 1

def contains(range_, point):
	r = Range(range_)
	return point >= r.start and point <= r.end

def allPoints(range_):
	r = Range(range_)
	return array([i + r.start for i in range(r.end - r.start + 1)], 'i')
	
def containsRange(range1_, range2_):
    for point in allPoints(range2_):
        if not contains(range1_, point):
            return False
    return True

def overlaps(range1_, range2_):
	return Range(range2_).start <= Range(range1_).end

File: kata/range/jruby/range.rb
class Range

	attr_accessor :start, :end

	def initialize(range)
		c = range.index ","
		@start = range[1..c].to_i
		if range[0] != "["
			@start = @start + 1
		end
		@end = range[c+1..range.length-2].to_i
		if range[range.length-1] != "]"
			@end = @end - 1
		end
	end

	def contains (point)
		point >= @start && point <= @end
	end
	
	def allPoints ()
		(Array.new(@end - @start+1) { |point| point + @start }).to_java Java::int
	end

	def containsRange (range)
		cc = true
		Range.new(range).allPoints.each { |point| cc = cc && contains(point) }
		cc
	end

	def overlaps (range)
		Range.new(range).start <= @end
	end
end



def contains (range, point)
	Range.new(range).contains point
end

def allPoints (range)
	Range.new(range).allPoints
end

def containsRange (range1, range2)
	Range.new(range1).containsRange range2
end

def overlaps (range1, range2)
	Range.new(range1).overlaps range2
end
